core/bin/external_node/src/main.rs

```diff
@@ -359,44 +387,189 @@ async fn init_tasks(
+                .with_polling_interval(config.optional.polling_interval())
+                .with_tx_sender(tx_sender)
+                .with_vm_barrier(vm_barrier)
+                .with_tree_api(tree_reader)
+                .enable_api_namespaces(config.optional.api_namespaces());
+
+        if let Some(sync_state) = sync_state {
+            builder = builder.with_sync_state(sync_state);
+        }
+
+        if let Some(ask_sync_state_from) = ask_sync_state_from {
+            builder = builder.with_ask_sync_state(ask_sync_state_from);
+        }
+
+        let ws_server_handles = builder
             .build()
             .context("failed to build WS JSON-RPC server")?
             .run(stop_receiver.clone())
             .await
             .context("Failed initializing WS JSON-RPC server")?;
+        app_health.insert_component(ws_server_handles.health_check);
+        task_futures.extend(ws_server_handles.tasks);
+    }
+
+    task_futures.extend(cache_update_handle);
+    task_futures.push(proxy_cache_updater_handle);
+
+    Ok(())
+}
+
+async fn init_tasks(
+    config: &ExternalNodeConfig,
+    connection_pool: ConnectionPool<Core>,
+    main_node_client: HttpClient,
+    task_handles: &mut Vec<task::JoinHandle<anyhow::Result<()>>>,
+    app_health: &AppHealthCheck,
+    stop_receiver: watch::Receiver<bool>,
+    components: &[Component],
+) -> anyhow::Result<()> {
+    let release_manifest: serde_json::Value = serde_json::from_str(RELEASE_MANIFEST)
+        .expect("releuse manifest is a valid json document; qed");
+    let release_manifest_version = release_manifest["core"].as_str().expect(
+        "a release-please manifest with \"core\" version field was specified at build time; qed.",
+    );
+
+    let version = semver::Version::parse(release_manifest_version)
+        .expect("version in manifest is a correct semver format; qed");
+    let pool = connection_pool.clone();
+    task_handles.push(tokio::spawn(async move {
+        loop {
+            let protocol_version = pool
+                .connection()
+                .await
+                .unwrap()
+                .protocol_versions_dal()
+                .last_used_version_id()
+                .await
+                .map(|version| version as u16);
+
+            EN_METRICS.version[&(format!("{}", version), protocol_version)].set(1);
 
-    app_health.insert_component(ws_server_handles.health_check);
-    app_health.insert_component(http_server_handles.health_check);
+            tokio::time::sleep(Duration::from_secs(10)).await;
+        }
+    }));
+
+    let singleton_pool_builder = ConnectionPool::singleton(&config.postgres.database_url);
+
+    // Run the components.
+    let tree_pool = singleton_pool_builder
+        .build()
+        .await
+        .context("failed to build a tree_pool")?;
+
+    let fee_params_fetcher = Arc::new(MainNodeFeeParamsFetcher::new(main_node_client.clone()));
```

Comment:
nit: this line looks weirdly out of place here, in the middle of tree initialization doe